locust/runners.py

```diff
@@ -346,11 +346,8 @@ def shape_worker(self) -> None:
             elif self.shape_last_tick == current_tick:
                 gevent.sleep(1)
             else:
-                if len(current_tick) == 2:
-                    user_count, spawn_rate = current_tick  # type: ignore
-                    user_classes = None
-                else:
-                    user_count, spawn_rate, user_classes = current_tick  # type: ignore
+                shape_adjustment_start = time.time()
+                user_count, spawn_rate, user_classes = current_tick if len(current_tick) == 3 else (*current_tick, None)  # type: ignore
```

Comment:
I think this is less readable than the old way. Does it really need to change or is it unrelated to the issue you are trying to fix?

Reply:
You're right, it's not related. I wanted to make it more "pythonic" and readable with my previous change (where I modified `current_tick` when its length was 2), but eventually it indeed got not very clear :) I restored the code from `master` for this part. Also, I changed a bit the `gevent.sleep` usage, to call it once for entire `tick()` handling - seems more understable for me now